staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -93,10 +93,10 @@ type Controller struct {
 	serviceQueue workqueue.RateLimitingInterface
 	nodeQueue    workqueue.RateLimitingInterface
 	// lastSyncedNodes is used when reconciling node state and keeps track of
-	// the last synced set of nodes. This field is concurrently safe because the
-	// nodeQueue is serviced by only one go-routine, so node events are not
-	// processed concurrently.
-	lastSyncedNodes []*v1.Node
+	// the last synced set of nodes per service key. This is accessed from the
+	// service and node controllers, hence it is protected by a lock.
+	lastSyncedNodes     map[string][]*v1.Node
```

Comment:
We can address this in a follow up of course, but IMO this is fine. The only services used as keys in this map are `type: LoadBalancer` - so it could be considered "a lot more information" on clusters which have a lot of services like that, but usually that's not the case.